### PR TITLE
Add foreman-systemd module

### PR DESCRIPTION
**Pull request by tomaszdurka**
https://github.com/ddollar/foreman/
For Debian 8.

---
Plus a convenient wrapper `foreman-systemd` for systemd:

**install**:
```
foreman-systemd install --user <user> <app-name> <Procfile-path>
```
Which runs:
```sh
# Uninstall:
rm -rf /etc/systemd/system/app-<app-name>*

# Install:
foreman export --user <user> --app app-<app-name> --root <Procfile-dir> systemd /etc/systemd/system
```

**start**:
```
foreman-systemd start <app-name>
```
Which runs:
```sh
systemctl start app-<app-name>.target
systemctl enable app-<app-name>.target
```

**stop**:
```
foreman-systemd stop <app-name>
```
Which runs:
```sh
systemctl disable app-<app-name>.target
systemctl stop app-<app-name>.target
```

**reload**:
```
foreman-systemd reload --user <user> <app-name> <Procfile-path>
```
Which runs:
```sh
stop()
install()
start()
```

### CODE REVIEW

**Comment by tomaszdurka** [comment]
@njam please review

File: modules/foreman_systemd/templates/binary.sh [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  echoerr "Usage:"
+  echoerr "  foreman-systemd install [-u <user>] <app-name> <procfile-path>"
+  echoerr "  foreman-systemd start <app-name>"
+  echoerr "  foreman-systemd stop <app-name>"
+  echoerr "  foreman-systemd reload [-u <user>] <app-name> <procfile-path>"
+  exit 1; 
+}
+
+install () {
+  uninstall $2
```
Comment (njam) [inline modules/foreman_systemd/templates/binary.sh]: Quotes needed around $2?

File: modules/foreman_systemd/templates/binary.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  
+
+case "${COMMAND}" in
+  install|reload)
+    if [ "$1" == "" ]; then
+      echoerr "Missing <profile-path> param"
+      echoerr
+      usage
+    fi
+    ROOT_DIR=$(dirname $1)s
```
Comment (njam) [inline modules/foreman_systemd/templates/binary.sh]: What is this "s"?

**Comment by njam** [comment]
See review in code.
Rest lgtm.
Did you test it?

File: modules/foreman_systemd/templates/binary.sh (hunk 2 above)
Comment (tomaszdurka) [inline modules/foreman_systemd/templates/binary.sh]: type

**Comment by tomaszdurka** [comment]
Did manual testing with formation:

**Procfile:**
```
foo: bash -c "while true; do echo foo; sleep 1; done"
bar: bash -c "while true; do echo bar; sleep 1; done"
```

Output
```
root@debian-8-amd64-plain:/home/vagrant# foreman-systemd reload -u root -f foo=4,bar=0 zoo /tmp
Removed symlink /etc/systemd/system/multi-user.target.wants/app-zoo.target.
[foreman export] cleaning up: /etc/systemd/system/app-zoo.target
[foreman export] cleaning up: /etc/systemd/system/app-zoo-bar.target
[foreman export] cleaning up: /etc/systemd/system/app-zoo-foo.target
[foreman export] cleaning up: /etc/systemd/system/app-zoo-bar@.service
[foreman export] cleaning up: /etc/systemd/system/app-zoo-foo@.service
[foreman export] cleaning up directory: /etc/systemd/system/app-zoo-foo.target.wants
[foreman export] cleaning up directory: /etc/systemd/system/app-zoo-bar.target.wants
[foreman export] writing: app-zoo-foo@.service
[foreman export] creating: app-zoo-foo.target.wants
[foreman export] symlinking: app-zoo-foo.target.wants/app-zoo-foo@5000.service -> ../app-zoo-foo@.service
[foreman export] symlinking: app-zoo-foo.target.wants/app-zoo-foo@5001.service -> ../app-zoo-foo@.service
[foreman export] symlinking: app-zoo-foo.target.wants/app-zoo-foo@5002.service -> ../app-zoo-foo@.service
[foreman export] symlinking: app-zoo-foo.target.wants/app-zoo-foo@5003.service -> ../app-zoo-foo@.service
[foreman export] writing: app-zoo-foo.target
[foreman export] writing: app-zoo-bar@.service
[foreman export] creating: app-zoo-bar.target.wants
[foreman export] writing: app-zoo-bar.target
[foreman export] writing: app-zoo.target
Created symlink from /etc/systemd/system/multi-user.target.wants/app-zoo.target to /etc/systemd/system/app-zoo.target.
root@debian-8-amd64-plain:/home/vagrant# systemctl list-dependencies app-zoo.target
app-zoo.target
● ├─app-zoo-bar.target
● └─app-zoo-foo.target
●   ├─app-zoo-foo@5000.service
●   ├─app-zoo-foo@5001.service
●   ├─app-zoo-foo@5002.service
●   └─app-zoo-foo@5003.service
```

**Comment by njam** [comment]
So it's actually doing the cleanup? "cleaning up: …"
Can you verify it, and if so we can remove our `rm -rf`…

**Comment by tomaszdurka** [comment]
Closing this as we seem to do fine with simple foreman/systemctl commands.

We can reopen if needed.

**Comment by kris-lab** [comment]
👍

**Pull request closed without merging**